cfme/physical/physical_storage.py

```diff
@@ -0,0 +1,231 @@
+import attr
+from navmazing import NavigateToAttribute
+from cached_property import cached_property
+
+from cfme.common import PolicyProfileAssignable, Taggable
+from cfme.common.physical_storage_views import (
+    PhysicalStorageDetailsView,
+    PhysicalStoragesView
+)
+from cfme.exceptions import ItemNotFound
+from cfme.modeling.base import BaseEntity, BaseCollection
+from cfme.utils.appliance.implementations.ui import CFMENavigateStep, navigate_to, navigator
+from cfme.utils.log import logger
+from cfme.utils.pretty import Pretty
+from cfme.utils.providers import get_crud_by_name
+from cfme.utils.update import Updateable
+from cfme.utils.wait import wait_for
+
+
+@attr.s
+class PhysicalStorage(BaseEntity, Updateable, Pretty, PolicyProfileAssignable, Taggable):
+    """Model of an Physical Storage in cfme.
+
+    Args:
+        name: Name of the physical Storage.
+
+    Usage:
+
+        mystorage = PhysicalStorage(name='sample_Storage')
+        mystorage.create()
+
+    """
+    pretty_attrs = ['name']
+
+    name = attr.ib()
+    provider = attr.ib(default=None)
+
+    INVENTORY_TO_MATCH = ['product_name', 'serial_number', 'description',
+                          'health_state', 'drive_bays', 'enclosure_count']
+
+    def load_details(self, refresh=False):
+        """To be compatible with the Taggable and PolicyProfileAssignable mixins.
+
+        Args:
+            refresh (bool): Whether to perform the page refresh, defaults to False
+        """
+        view = navigate_to(self, "Details")
+        if refresh:
+            view.browser.refresh()
+            view.flush_widget_cache()
+
+    def _execute_button(self, button, option, handle_alert=False):
+        view = navigate_to(self, "Details")
+        view.toolbar.custom_button(button).item_select(option, handle_alert=handle_alert)
+        return view
+
+    def _execute_action_button(self, button, option, handle_alert=True, **kwargs):
+        target = kwargs.get("target", None)
+        provider = kwargs.get("provider", None)
+        desired_state = kwargs.get("desired_state", None)
+
+        view = self._execute_button(button, option, handle_alert=handle_alert)
+
+        if desired_state:
+            self._wait_for_state_change(desired_state, target, provider, view)
+        elif handle_alert:
+            wait_for(
+                lambda: view.flash.is_displayed,
+                message="Wait for the handle alert to appear...",
+                num_sec=5,
+                delay=2
+            )
+
+    def product_name(self):
+        view = navigate_to(self, "Details")
+        return view.entities.properties.get_text_of("Product Name")
+
+    def serial_number(self):
+        view = navigate_to(self, "Details")
+        return view.entities.properties.get_text_of("Serial Number")
+
+    def health_state(self):
+        view = navigate_to(self, "Details")
+        return view.entities.properties.get_text_of("Health State")
+
+    def enclosure_count(self):
```

Comment:
Optional:
If those are not going to use repeatedly in your test then no need to add in a model page. You can easily use in a test whenever you want. The model page should be standard and effective methods or property.

Reply:
**Required**

I don't like this pattern of trying to reflect the UI summary table values as entity properties.  Please remove from this PR scope, or include and let's discuss some better options to reduce all this boilerplate.

This could be done with a `_getattr_` implementation that catches attributes that are detail properties and returns them from the UI, and otherwise resolves `super().__getattr__()`